examples/structured_data/advanced_techniques_for_structured_data.py

```diff
@@ -0,0 +1,472 @@
+to handle input features, and encoding categorical features using one-hot encoding and
+embedding respesentation.
+
+
+Note that this example should be run with TensorFlow 2.3 or higher.
+"""
+
+"""
+## The dataset
+
+This example uses the [covertype](https://archive.ics.uci.edu/ml/datasets/covertype) from UCI
+Machine Learning Repository. The task is to Predict forest cover type from cartographic variables.
+The dataset includes 506,011 intances with 12 input features, 10 of which numerical, and the other 2 are
+categorical. The target class has 7 labels.
+"""
+
+"""
+## Setup
+"""
+
+import tensorflow as tf
+import pandas as pd
+from tensorflow import keras
+from tensorflow.keras import layers
+import math
+from sklearn.model_selection import train_test_split
+
+"""
+## Prepare the data
+
+First, let's download the data from the UCI Machine Learning Repository to a pandas Dataframe:
+"""
+
+data_url = (
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/covtype/covtype.data.gz"
+)
+raw_data = pd.read_csv(data_url, header=None)
+print(f"Dataset shape: {raw_data.shape}")
+raw_data.head()
+
+"""
+The two catrogircal features in the dataset are binary encoded.
+We will convert this dataset repesentation to the typical repesentation, where each categorical
+feature is repesented as on column.
+"""
+
+soil_type_values = [f"soil_type_{idx+1}" for idx in range(40)]
+wilderness_area_values = [f"area_type_{idx+1}" for idx in range(4)]
+
+soil_type = raw_data.loc[:, 14:53].apply(
+    lambda x: soil_type_values[0::1][x.to_numpy().nonzero()[0][0]], axis=1
+)
+wilderness_area = raw_data.loc[:, 10:13].apply(
+    lambda x: wilderness_area_values[0::1][x.to_numpy().nonzero()[0][0]], axis=1
+)
+
+CSV_HEADER = [
+    "Elevation",
+    "Aspect",
+    "Slope",
+    "Horizontal_Distance_To_Hydrology",
+    "Vertical_Distance_To_Hydrology",
+    "Horizontal_Distance_To_Roadways",
+    "Hillshade_9am",
+    "Hillshade_Noon",
+    "Hillshade_3pm",
+    "Horizontal_Distance_To_Fire_Points",
+    "Wilderness_Area",
+    "Soil_Type",
+    "Cover_Type",
+]
+
+data = pd.concat(
+    [raw_data.loc[:, 0:9], wilderness_area, soil_type, raw_data.loc[:, 54]],
+    axis=1,
+    ignore_index=True,
+)
+data.columns = CSV_HEADER
+
+# Change the target label indicies to range from 0 to 6.
+data["Cover_Type"] = data["Cover_Type"] - 1
+
+print(f"Dataset shape: {data.shape}")
+data.head().T
+
+"""
+You can see now the Dataframe has 13 columns per sample (12 features, plus the target label).
+
+Let's split the data into training and testing splits, with 85% and 15% of the instance, respectively.
+"""
+
+train_data, test_data = train_test_split(
+    data, train_size=0.85, stratify=data["Cover_Type"]
+)
+
+print(f"Train split size: {len(train_data.index)}")
+print(f"Test split size: {len(test_data.index)}")
+
+"""
+Now we store the train and test data splits to CSV files.
```

Comment:
Suggestion:
- "Training" data (for consistency)

```suggestion
Next, store the training and test data in separate CSV files.
```